File: app/src/main/java/br/edu/ifsp/scl/sdm/dicesdm/MainActivity.java
package br.edu.ifsp.scl.sdm.dicesdm;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    // Gerador de números randômicos usado para simular o lançamento do dado
    private Random geradorRandomico;

    // Referências para as Views do arquivo de layout
    private TextView resultadoTextView;
    private ImageView resultadoImageView;
    private Spinner numDicesSpinner;
    private ImageView resultado2ImageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Após a criação da tela
        geradorRandomico = new Random(System.currentTimeMillis());

        // Recuperando referência para as Views do arquivo de layout
        resultadoTextView = findViewById(R.id.resultadoTextView);
        resultadoImageView = findViewById(R.id.resultadoImageView);
        numDicesSpinner = findViewById(R.id.numDicesSpinner);
        resultado2ImageView = findViewById(R.id.resultado2imageView);
    }

    public void jogarDado(View view){
        if (view.getId() == R.id.jogarDadoButton) {
            int numDices = Integer.parseInt(numDicesSpinner.getSelectedItem().toString());
            if (numDices == 2) {
                resultado2ImageView.setVisibility(View.VISIBLE);
            }
            else{
                resultado2ImageView.setVisibility(View.GONE);
            }

            String resultadoText = "";
            for (int i = 1; i <= numDices; i++) {
                int resultado = geradorRandomico.nextInt(6) + 1;

                resultadoText += resultado + " ";

                ImageView imageView = i == 1? resultadoImageView : resultado2ImageView;
                switch (resultado) {
                    case 1:
                        imageView.setImageResource(R.drawable.dice_1);
                        break;
                    case 2:
                        imageView.setImageResource(R.drawable.dice_2);
                        break;
                    case 3:
                        imageView.setImageResource(R.drawable.dice_3);
                        break;
                    case 4:
                        imageView.setImageResource(R.drawable.dice_4);
                        break;
                    case 5:
                        imageView.setImageResource(R.drawable.dice_5);
                        break;
                    case 6:
                        imageView.setImageResource(R.drawable.dice_6);
                        break;
                }
            }

            resultadoTextView.setText(getString(R.string.a_face_sorteada_foi) + resultadoText);
        }
    }
}

File: app/src/main/java/br/edu/ifsp/scl/sdm/dicesdm/MainActivity.kt
package br.edu.ifsp.scl.sdm.dicesdm

import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.ImageView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    val geradorRandomico: Random
    init {
        geradorRandomico = Random(System.currentTimeMillis())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun jogarDado(view: View) {
        if (view == jogarDadoButton) {
            val numDices:Int = numDicesSpinner.selectedItem.toString().toInt()
            val numFaces = numFacesEditText.text.toString().toInt()

            if (numFaces > 6) {
                resultadoImageView.visibility = View.GONE
                resultado2ImageView.visibility = View.GONE
            } else {
                resultadoImageView.visibility = View.VISIBLE
                resultado2ImageView.visibility = if (numDices == 2) View.VISIBLE else View.GONE
            }

            var resultadoText = ""
            for (i in 1..numDices) {
                val resultado = geradorRandomico.nextInt(numFaces) + 1
                resultadoText = "$resultadoText $resultado"

                val imageView: ImageView = if (i==1) resultadoImageView else resultado2ImageView
                val resourceName: String = "dice_${resultado}"
                imageView.setImageResource(resources.getIdentifier(resourceName,"drawable",packageName))
                /*when (resultado) {
                    1  -> imageView.setImageResource(R.drawable.dice_1)
                    2  -> imageView.setImageResource(R.drawable.dice_2)
                    3  -> imageView.setImageResource(R.drawable.dice_3)
                    4  -> imageView.setImageResource(R.drawable.dice_4)
                    5  -> imageView.setImageResource(R.drawable.dice_5)
                    6  -> imageView.setImageResource(R.drawable.dice_6)
                }*/

                resultadoTextView.text = "A face sorteada foi: ${resultadoText}"
            }
        }
    }
}